rust/connlib/shared/src/callbacks.rs

```diff
@@ -39,6 +42,169 @@ impl From<Ipv6Network> for Cidrv6 {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum Status {
+    Unknown,
+    Online,
+    Offline,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+#[serde(tag = "type", rename_all = "snake_case")]
```

Comment:
Do these need to be serializable? With whom do we have a contract here?

Reply:
I think it's needed for crossing the FFI boundary on phones and the IPC boundary on Windows and Linux.

Reply:
yeah, that's exactly it